classes/sota_sanity.bbclass

```diff
@@ -10,6 +10,55 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    import re
+    import os.path
+
+# Translate old provisioning recipe names into the new versions.
+#    prov = d.getVar("SOTA_CLIENT_PROV")
+#   if prov == "aktualizr-auto-prov":
+#        bb.warn('aktualizr-auto-prov is deprecated. Please use aktualizr-shared-prov instead.')
+#        d.setVar("SOTA_CLIENT_PROV", "aktualizr-shared-prov")
+#    elif prov == "aktualizr-ca-implicit-prov":
+#        bb.warn('aktualizr-ca-implicit-prov is deprecated. Please use aktualizr-device-prov instead.')
+#        d.setVar("SOTA_CLIENT_PROV", "aktualizr-device-prov")
+#    elif prov == "aktualizr-hsm-prov":
+#        bb.warn('aktualizr-hsm-prov is deprecated. Please use aktualizr-device-prov-hsm instead.')
+#        d.setVar("SOTA_CLIENT_PROV", "aktualizr-device-prov-hsm")
+
+    if d.getVar("OSTREE_BRANCHNAME") and re.match("^[a-zA-Z0-9_-]*$", d.getVar("OSTREE_BRANCHNAME")) is None:
+        status.addresult("OSTREE_BRANCHNAME Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if d.getVar("SOTA_HARDWARE_ID") and re.match("^[a-zA-Z0-9_-]*$", d.getVar("SOTA_HARDWARE_ID")) is None:
+        status.addresult("SOTA_HARDWARE_ID Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if d.getVar("SOTA_CLIENT_FEATURES") and d.getVar("SOTA_CLIENT_FEATURES") not in ("hsm", "secondary-network", ""):
```

Comment:
This is cool, but I just realized that it might not work if `SOTA_CLIENT_FEATURES` includes both possible options. We might need to split it into tokens and verify that each token is in the expected list of features.

Reply:
Not sure if I’ getting this right, you mean SOTA_CLIENT_FEATURES = “ham secondary-network” might not work?

Reply:
Yep!

Reply:
But isn't "hsm secondary-network" or "secondary-network hsm" not in ("hsm", "secondary-network", "")?

Reply:
I searched `SOTA_CLIENT_FEATURES` on GitHub found `serialcan` `ubootenv` might also be possible options.

Reply:
Good catch. I just did some searching as well and realized that `secondary-network` is no longer used. Would you mind making a PR to update the docs? They live in the aktualizr repo now (`docs/ota-client-guide/modules/ROOT/pages/build-configuration.adoc`).

Reply:
Of course, but how should I describe `serialcan` and `ubootenv`?

Reply:
Good point. Let me look into it and I'll open a PR.